### PR TITLE
Adding ADF V2 SDK

### PR DESCRIPTION
**Pull request by bgold09**
## Description
  * This SDK is for the new Azure Data Factory V2 platform/service,
    which lives side-by-side with the existing "V1" GA service.
  * code generated based on https://github.com/Azure/azure-rest-api-specs/pull/1703

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md).**
- [x] **The pull request does not introduce [breaking changes](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/Documentation/breaking-changes.md).**

### [General Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#general-guidelines)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/dev/documentation/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#testing-guidelines)
- [x] Pull request includes test coverage for the included changes.

### [SDK Generation Guidelines](https://github.com/Azure/azure-sdk-for-net/blob/AutoRest/.github/CONTRIBUTING.md#sdk-generation-guidelines)
- [x] If an SDK is being regenerated based on a new swagger spec, a link to the pull request containing these swagger spec changes has been included above.
- [x] The generate.cmd file for the SDK has been updated with the version of AutoRest, as well as the commitid of your swagger spec or link to the swagger spec, used to generate the code.
- [x] The `*.csproj` and `AssemblyInfo.cs` files have been updated with the new version of the SDK.

### CODE REVIEW

File: src/SDKs/DataFactory/Management.DataFactory/Properties/AssemblyInfo.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Reflection;
+using System.Resources;
+
+[assembly: AssemblyTitle("Microsoft Azure Data Factory Management Library")]
+[assembly: AssemblyDescription("Provides management functionality for Microsoft Azure Data Factory Resources.")]
+[assembly: AssemblyVersion("0.0.0.0")]
```
Comment (shahabhijeet) [inline src/SDKs/DataFactory/Management.DataFactory/Properties/AssemblyInfo.cs]: Assembly version cannot be 0.0.0.0 While you are in preview start with 0.2.0.0
Moving forward, bump up build version in AssemblyFileVersion to  0.2.1.0 and 0.2.2.0 and so on and leave AssemblyVersion at 0.2.0.0

File: src/SDKs/DataFactory/Management.DataFactory/Microsoft.Azure.Management.DataFactory.csproj [hunk 2]
```diff
@@ -0,0 +1,13 @@
+﻿<Project Sdk="Microsoft.NET.Sdk">
+  <Import Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.reference.props'))" />
+  <PropertyGroup>
+    <PackageId>Microsoft.Azure.Management.DataFactory</PackageId>
+    <Description>Azure Data Factory v2 is a fully managed hybrid data service to compose and manage data integration workflows at scale that are reliable and secure. Use native ADF data connectors and Integration Runtimes to move and transform cloud and on-premises data that can be unstructured, semi-structured, and structured with Hadoop, Azure Data Lake, Spark, SQL Server, Cosmos DB and many other data platforms.</Description>
+    <VersionPrefix>0.2.0-preview</VersionPrefix>
+    <AssemblyName>Microsoft.Azure.Management.DataFactory</AssemblyName>
+    <PackageTags>Microsoft Azure resource management;Data Factory;ADF;REST HTTP client;azureofficial;windowsazureofficial;netcore451511</PackageTags>
```
Comment (shahabhijeet) [inline src/SDKs/DataFactory/Management.DataFactory/Microsoft.Azure.Management.DataFactory.csproj]: Add < PackageReleaseNotes > and add Release notes.
I would recommend to use < ![CDATA and then format your Release notes

https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/MarketplaceOrdering/Management.MarketplaceOrdering/Microsoft.Azure.Management.MarketplaceOrdering.csproj#L10

File: src/SDKs/DataFactory/DataFactory.Tests/DataFactory.Tests.csproj [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      <CopyToOutputDirectory>PreserveNewest</CopyToOutputDirectory>
+    </None>
+  </ItemGroup>
+
+  <ItemGroup>
+    <Service Include="{82a7f48d-3b50-4b1e-b82e-3ada8210c358}" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <PackageReference Update="Microsoft.NET.Test.Sdk" Version="15.3.0" />
```
Comment (shahabhijeet) [inline src/SDKs/DataFactory/DataFactory.Tests/DataFactory.Tests.csproj]: You need to re record tests using the latest version of Resource Manager at least 1.6.0

**Review by shahabhijeet — CHANGES_REQUESTED** [review]

**Review by shahabhijeet — CHANGES_REQUESTED** [review]
Have a very clear description as to what is the difference between nuget package that ends with DataFactories
and
DataFactory (this version)

File: src/SDKs/DataFactory/DataFactory.Tests/DataFactory.Tests.csproj (hunk 3 above)
Comment (lygasch) [reply src/SDKs/DataFactory/DataFactory.Tests/DataFactory.Tests.csproj]: Brian, who has been working on the C#, is on vacation and neither Hermine nor I understand this comment in his PR that we are trying to complete. Can you help us learn exactly what needs to be done to address it? And is it a blocker for this merge or can it wait?

Comment (lygasch) [reply src/SDKs/DataFactory/DataFactory.Tests/DataFactory.Tests.csproj]: Opened https://github.com/Azure/azure-sdk-for-net/issues/3726  and assigned to BrGold

File: src/SDKs/DataFactory/Management.DataFactory/Properties/AssemblyInfo.cs (hunk 1 above)
Comment (bgold09) [reply src/SDKs/DataFactory/Management.DataFactory/Properties/AssemblyInfo.cs]: done

**Comment by bgold09** [comment]
@shahabhijeet where would a description between DataFactories and DataFactory packages go?

File: src/SDKs/DataFactory/Management.DataFactory/Microsoft.Azure.Management.DataFactory.csproj (hunk 2 above)
Comment (bgold09) [reply src/SDKs/DataFactory/Management.DataFactory/Microsoft.Azure.Management.DataFactory.csproj]: In the past, we have written our own [changelog](https://github.com/Azure/azure-sdk-for-net/blob/master/src/ResourceManagement/DataFactory/DataFactoryManagement/DataFactoryManagementChangelog.md) to serve this purpose and put a link to a [doc page](https://github.com/Azure/azure-sdk-for-net/blob/master/src/ResourceManagement/DataFactory/DataFactoryManagement/Microsoft.Azure.Management.DataFactories.nuspec#L6) that we create with that content. 

Can we do a similar thing this time around? Right now we don't have a link to where that content will be for this new package.

**Comment by shahabhijeet** [comment]
Closing this PR as the other PR was merged
https://github.com/Azure/azure-sdk-for-net/pull/3727

**Pull request closed without merging**